src/commands/mod.rs

```diff
@@ -736,6 +774,372 @@ impl DaemonControl {
+        // We are assuming we only have one directly conflicting transaction.
+        if mempool_entry.descendant_count > RBF_MAX_DESCENDANT_COUNT {
+            return Err(CommandError::RbfError(RbfErrorInfo::TooManyDescendants(
+                mempool_entry.descendant_count,
+            )));
+        }
+        let prev_feerate_vb = {
+            let prev_abs_fee = prev_psbt
+                .fee()
+                .expect("fee must be Ok as transaction is already in mempool");
+            let prev_tx_size: u64 = prev_psbt
+                .clone()
+                .extract_tx()
+                .vsize()
+                .try_into()
+                .expect("vsize must fit in a u64");
+            prev_abs_fee
+                .checked_div(prev_tx_size)
+                .ok_or(CommandError::InsaneFees(InsaneFeeInfo::InvalidFeerate))?
+        };
+        // Check replacement transaction's feerate is greater than previous feerate (the only directly conflicting transaction).
+        // TODO: Should we throw error or just set feerate_vb = prev_feerate_vb + 1 and check `rbf_psbt` below?
+        if feerate_vb < prev_feerate_vb.to_sat().checked_add(1).unwrap() {
+            return Err(CommandError::RbfError(RbfErrorInfo::TooLowFeerate(
+                feerate_vb,
+            )));
+        }
+        let prev_txos = &prev_psbt.unsigned_tx.output;
+        let mut destinations: HashMap<bitcoin::Address, bitcoin::Amount> =
+            HashMap::with_capacity(prev_txos.len());
+        let mut change_index: Option<bip32::ChildNumber> = None;
+        for txo in prev_txos {
+            let address = bitcoin::Address::from_script(
+                &txo.script_pubkey,
+                self.config.bitcoin_config.network,
+            )
+            .map_err(CommandError::Address)?;
+            match db_conn.derivation_index_by_address(&address) {
+                Some((ind, true)) => {
+                    // TODO: check if there is more than one change output?
+                    println!("found change at address {} and index {}", address, ind);
+                    change_index = Some(ind);
+                }
+                _ => {
+                    // Non-change outputs are handled in the same way whether they are ours or not.
+                    if !is_cancel {
+                        destinations.insert(address, bitcoin::Amount::from_sat(txo.value));
+                    }
+                }
+            };
+        }
+        // We take all previous inputs as mandatory candidates and, if not self-send, include
+        // confirmed coins as optional.
+        // TODO: For `is_cancel`, shall we include just one previous input?
+        let mut candidate_coins: Vec<CandidateCoin> = prev_coins
+            .into_values()
+            .map(|c| CandidateCoin {
+                coin: c,
+                must_select: true,
+            })
+            .collect();
+        if !destinations.is_empty() {
```

Comment:
Why not including them if it's a self-spend?

Reply:
My thinking was that for a self-send, we only want to consider the same coins again, in the same way that we don't use coin selection for a new self-send. But perhaps for RBF, allowing to add an extra coin to cover the fee would make sense.

Reply:
Yes i think it should be expected that if you are going to RBF, some new inputs may be added to the transaction. We should warn the user in the documentation. And maybe also in the GUI but i'm not sure how not to confuse the user who doesn't care (most users don't i think).

Reply:
I added a comment to the API doc that additional coins may be used, including for self-sends.